browser/brave_talk/brave_talk_service.h

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_BRAVE_TALK_BRAVE_TALK_SERVICE_H_
+#define BRAVE_BROWSER_BRAVE_TALK_BRAVE_TALK_SERVICE_H_
+
+#include <string>
+
+#include "base/callback_forward.h"
+#include "components/keyed_service/core/keyed_service.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents_observer.h"
+
+namespace content {
+class WebContents;
+}
+
+namespace brave_talk {
+
+// Service for managing requests to |brave.beginAdvertiseShareDisplayMedia|. Note:
```

Comment:
@petemill what do you think of this approach/tradeoff? It's definitely the simplest, as it means we don't need to implement any new UI for picking which frame to share with, and least fraught. However, we definitely wouldn't be able to do this if we were planning on desigining an API for the wider web (any website could steal the tab share button, any time).